playwright/src/test/java/com/microsoft/playwright/TestBrowserTypeConnect.java

```diff
@@ -85,22 +91,24 @@ private static BrowserServer launchBrowserServer(BrowserType browserType) {
     }
   }
 
-  @Override
+  public static class ConnectOptionsFactory implements OptionsFactory {
+    @Override
+    public Options getOptions() {
+      //wsEndpoint is set dynamically based on what is returned from the BrowserServer
+      return new Options().setConnectOptions(new BrowserType.ConnectOptions().setSlowMo(1));
+    }
+  }
+
   @BeforeAll
-  // Hide base class method to launch browser server and connect to it.
-  void launchBrowser() {
-    initBrowserType();
+  static void launchBrowser(BrowserType browserType) {
     BrowserServer r = launchBrowserServer(browserType);
     wsEndpoint = r.wsEndpoint;
     browserServer = r.process;
-    browser = browserType.connect(wsEndpoint);
-    // Do not actually connect to browser, the tests will do it manually.
+    OptionsExtension.threadLocalOptions.set(new ConnectOptionsFactory().getOptions().setWsEndpoint(wsEndpoint));
```

Comment:
@yury-s this is why the `ThreadLocal<Options>` needed to be public.  The `wsEndpoint` isn't known until the `BeforeAll` hook is executed, at that point the `Options` object has already been created.  We needed a way to dynamically alter this.  

I am not advocating for this behavior, it's just what I saw as necessary.  We can make it private again and I can set this option value via reflection.

Reply:
I don't like us writing into variables which are very specific to the implementation of the fixtures. Can we instead lazily launch the browser server from inside `ConnectOptionsFactory.getOptions` when creating the `Options` instance?

Alternatively, we can leave existing `TestBrowserTypeConnect` test alone and create a new one under `./tools`. We could launch the browser server separately and then run a playwright test that populates `ws` address in UsePlaywright options with e.g. an environment variable value. We are testing a very specific setup with the remote connection and I believe having a dedicated test would be justified in this case. WDYT?

Reply:
> Alternatively, we can leave existing TestBrowserTypeConnect test alone and create a new one under ./tools. We could launch the browser server separately and then run a playwright test that populates ws address in UsePlaywright options with e.g. an environment variable value. We are testing a very specific setup with the remote connection and I believe having a dedicated test would be justified in this case. WDYT?

I like this but there is a lot going on in the `launchBrowser` methods that I don't totally understand.  I would need some help with how that would work.

But this actually gives me another idea.  We can create another internal extension to launch the browser and set a variable with the wsEndpoint.  I'll push up what I'm thinking for your review.

Reply:
Sure, I can help with that. Let's revert changes in this file and in OptionsExtension.java and land the rest without a test for now. We can add it later while letting you use it in the other project. We can also not add a test for now if it implies a lot of extra complexity. I don't think we have such test for Node.js, but we run it with the service, could do the same here.

Reply:
Sounds good to me.  I've reverted the tests and extension changes.